src/libs/ck-libs/io/ckio.C

```diff
@@ -389,6 +638,135 @@ namespace Ck { namespace IO {
         }
       };
 
+	      
+	/**
+	 * These are the designated readers that go to disk
+	 * and get the data. They are also responsible for holding on 
+	 * to the data and are who give copies of the data to 
+	 * the ReadAssemblers who need to satisfy read requests.
+	 * The number of BufferChares is configurable in the Options struct
+	 * when the user sets up their read session.
+	 */
+      class BufferChares : public CBase_BufferChares {
+	private:
+		FileToken _token; // the token of the given file
+      		const FileInfo* _file; // the pointer to the FileInfo
+		size_t _session_bytes; // number of bytes in the session
+		size_t _session_offset; // the offset of the session
+		size_t _my_offset;
+		size_t _my_bytes;
+		std::shared_future<char*> _buffer;
+		size_t _num_readers;	
+		size_t _read_stripe;
+	public:
+		BufferChares(FileToken file, size_t offset, size_t bytes, size_t num_readers) : _token(file), _file(CkpvAccess(manager)->get(file)), _session_bytes(bytes), _session_offset(offset){
+			_num_readers = num_readers;
+			_read_stripe = bytes / num_readers;
+			_my_offset = thisIndex * (_read_stripe) + _session_offset;
+			_my_bytes = min(_read_stripe, _session_offset + _session_bytes - _my_offset); // get the number of bytes owned by the session
+			// last BufferChares array; read the remaining stuff
+			if(thisIndex == _num_readers - 1) 
+				_my_bytes = _session_offset + _session_bytes - _my_offset;
+
+			CkAssert(_file -> fd != -1);
+			CkAssert(_my_offset >= _session_offset);
+			CkAssert(_my_offset + _my_bytes <= _session_offset + _session_bytes);
+			#ifdef DEBUG
+			CkPrintf("Inside constructor of BufferChares[%d]; I own %zu bytes starting from %zu offset. about o start the readData function\n", thisIndex, _my_bytes, _my_offset);
+			#endif
+			double disk_read_before = CkWallTimer(); // get the before disk_read
+			std::future<char*> temp_buffer = std::async(std::launch::async, &BufferChares::readData, this);
+			_buffer = temp_buffer.share();
+			double disk_read_after = CkWallTimer(); // end disk time
+			double total_time = disk_read_after - disk_read_before;
+			CkCallback cb(CkReductionTarget(BufferChares, printTime), thisProxy[0]);
+			contribute(sizeof(double), &total_time, CkReduction::max_double, cb);
+		}
+		// deprecated function; not used any longer
```

Comment:
If it's deprecated, we should remove it.

Reply:
done!